api/base/parsers.py

```diff
@@ -74,6 +75,14 @@ def flatten_data(self, resource_object, parser_context, is_list):
             if object_type is None:
                 raise JSONAPIException(source={'pointer': '/data/type'}, detail=NO_TYPE_ERROR)
 
+        if parser_context['request'].version >= 2.6:
+            if 'preprint' in parser_context['request'].path and request_method == 'PATCH':
+                if object_id is None:
+                    raise JSONAPIException(source={'pointer': '/data/id'}, detail=NO_ID_ERROR)
+
+                if object_type is None:
+                    raise JSONAPIException(source={'pointer': '/data/type'}, detail=NO_TYPE_ERROR)
```

Comment:
Please also check if the object type exists and matches one of the accepted types for the current view (see comment below).